### PR TITLE
Beanie

### PR DESCRIPTION
**Pull request by lmarini**
Example of how to use benie to list datasets (using mongo views). There is new objects for `DatasetsDB` and `AuthorizationDB`. Both have been replaced in most places (not all). I left the old code there for comparison but it should be removed before it is merged.

*I couldn't figure out a way to return `id` from fastapi instead of `_id` so for now we replace `_id` with `id` in the javascript after the ajax call. If we decided to adopt benie we could change frontend to use `_id`.*

To test migrations you have to provide database flag with `-db`. If you put it in the URI it will not work:

`beanie migrate -uri "mongodb://localhost" -db clowder2 -p app/models/migrations`

Migrations are explained here https://beanie-odm.dev/tutorial/migrations/.

### CODE REVIEW

**Review by longshuicy — COMMENTED** [review]
For some reason I cannot see any datasets. Do i need to drop the database?
image

File: frontend/src/actions/dataset.js [hunk 1]
```diff
@@ -129,9 +129,11 @@ export function fetchDatasets(skip=0, limit=21, mine=false){
 		// TODO: Parameters for dates? paging?
 		return V2.DatasetsService.getDatasetsApiV2DatasetsGet(skip, limit, mine)
 			.then(json => {
+				// FIXME temporary workaround to map from `_id` returned by API to `id` expected by javascript
```
[the remark this answers is not in the document]
Comment (lmarini) [reply frontend/src/actions/dataset.js:132]: You have to create new ones. The new datasets and authorization objects go in new collections. I did that just to make it easier to test it out.

**Comment by lmarini** [comment]
I can't get `metadata_files.get_file_metadata()` to work because of the way we are creating the variable arguments query. Taking a break from it but didn't want to forget so putting it here.

**Comment by lmarini** [comment]
Unable to add group to dataset. Looks like the group id is null when making rest call. Looking at frontend we might not be setting the state properly, see `ShareGroupDatasetModal:106`.

**Comment by lmarini** [comment]
> Unable to add group to dataset. Looks like the group id is null when making rest call. Looking at frontend we might not be setting the state properly, see `ShareGroupDatasetModal:106`.

Fixed, manual cast to GroupOut.

**Comment by lmarini** [comment]
Edit medata fails with 
```  File "/Users/lmarini/code/clowder2/backend/app/routers/metadata_datasets.py", line 218, in update_dataset_metadata
    query.append(MetadataDB.definition == definition)
AttributeError: 'dict' object has no attribute 'append'```

Looks like we are mixing query definitions methods, line 196.

**Comment by ddey2** [comment]
Seeing the following error during extraction and viewing User metadata and Extraction Metadata tab
Screenshot 2023-06-05 at 6 54 53 AM
Screenshot 2023-06-05 at 6 47 12 AM
Screenshot 2023-06-05 at 6 47 23 AM

File: backend/app/routers/groups.py [hunk 2]
```diff
@@ -48,32 +44,22 @@ async def get_groups(
...
-        .to_list(length=limit)
-    ):
-        groups.append(GroupOut.from_mongo(doc))
-    return groups
+    groups = await GroupDB.find(
+        Or(
+            GroupDB.creator == user_id,
+            GroupDB.users.user.email == user_id,
+        ),
+        sort=("created", DESCENDING),
```
Comment (max-zilla) [inline backend/app/routers/groups.py:52]: might also see `sort=(-GroupDB.created)`

**Review by max-zilla — APPROVED** [review]

**Pull request merged**